core/logger/logger.go

```diff
@@ -104,10 +107,17 @@ func copyFields(fields []interface{}, add ...interface{}) []interface{} {
 	return f
 }
 
+func joinName(old, new string) string {
+	if old == "" {
+		return new
+	}
+	return old + "." + new
+}
+
 func (l *zapLogger) Named(name string) Logger {
 	newLogger := *l
-	newLogger.SugaredLogger = l.SugaredLogger.Named(name).With("id", name)
-	newLogger.fields = copyFields(l.fields, "id", name)
+	newLogger.name = joinName(l.name, name)
+	newLogger.SugaredLogger = l.SugaredLogger.Named(name)
```

Comment:
I suspect that `With("id"` was only being used to work around the name not being passed along, and that it was OK to drop as redundant now. We still have control of the `logger` name key, if we would prefer it to be `id` instead, but that seemed more error prone to accidental overwrites from name clashes.

Reply:
This is actually more widespread than I initially thought, so `id` should probably stay. Then the question is whether to use the full (zap)name as `id=a.b.c`, or to keep both `id=c` and `logger=a.b.c`. (_Or_ to just ditch the sub-scoping from zap and do only `id=c` - we only use one level currently anyways, and this would simplify our implementation and further decouple from zap)

Reply:
Restored to keep both.

Reply:
Circled back to dropping `id` again because zap doesn't support overwriting keys. So we just have `logger=a.b.c`.